src/index.ts

```diff
@@ -32,3 +33,49 @@ export async function importResourceType(resourceName: string, _resourceVersion:
 
   return type.TypeName;
 };
+
+export interface ImportProductOptions extends DescribeProductAggregateOptions {
+  /**
+   * @default "./sc-products"
+   */
+  readonly outdir?: string;
+}
+
+/**
+ * Entry point to import Service Catalog product resource.
+ *
+ * @param outdir the out folder to use (defaults to the current directory under a 'sc-products' folder)
+ * @returns name of the product version
+ */
+export async function importProduct(options: ImportProductOptions): Promise<string> {
+  const outdir = options.outdir ?? '.';
+
+  await describeProductAggregate(options);
+
+  //TODO CodeGen
+
+  return outdir; //just for typechecking
+};
+
+/**
+ * Entry point to import all available Service Catalog product resources with `DEFAULT` parameters.
+ *
+ * @param outdir the out folder to use (defaults to the current directory under a 'sc-products' folder)
+ * @returns names of the product versions
+ */
+export async function importProducts(options: ImportProductOptions): Promise<string[]> {
+  const outdir = options.outdir ?? '.';
+  let productVersions: string[] = [];
+
+  const availableProducts = await fetchAvailableProducts();
+
+  await Promise.all(availableProducts.map(async (product) => {
+    const productVersion = await importProduct( {
+      outdir: outdir,
+      productId: product.Id!,
```

Comment:
`product.Id` is always defined?

Reply:
Yes, it's a weird artifact from the SDK but our api will always return the ID.  We are going to wrap this handling anyway so that if one product out of _n_ fails that we don't fail the entire codegen but rather output that to the console alongside the succeeded products.